vermouth/molecule.py

```diff
@@ -388,18 +599,71 @@ def __init__(self):
         self.patterns = []
 
 
-def attributes_match(attributes, template_attributes):
+def attributes_match(attributes, template_attributes, ignore_keys=()):
+    """
+    Compare a dict of attributes from a molecule with one from a link.
+
+    Returns ``True`` if the attributes from the link match the ones from the
+    molecule; returns ``False`` otherwise. The attributes from a link match
+    with those of a molecule is all the individual attribute from the link
+    match the corresponding ones in the molecule. In the simplest case, these
+    attribute match if their values are equal. If the value of the link
+    attribute is an instance of :class:`LinkPredicate`, then the attributes
+    match if the ``match`` method of the predicate returns ``True``.
+
+    Parameters
+    ----------
+    attributes: dict
+        Attributes from the molecule.
+    template_attributes: dict
+        Attributes from the link.
+    ignore_keys: list
+        List of keys to ignore from 'template_attributes'.
+
+    Returns
+    -------
+    bool
+    """
     for attr, value in template_attributes.items():
-        if isinstance(value, Choice):
-            if attributes.get(attr) not in value:
-                return False
-        else:
-            if attributes.get(attr) != value:
+        if attr in ignore_keys:
+            continue
+        if isinstance(value, LinkPredicate):
+            match = value.match(attributes, attr)
+            if not value.match(attributes, attr):
                 return False
+        elif attributes.get(attr) != value:
+            return False
     return True
 
 
 def interaction_match(molecule, interaction, template_interaction):
+    """
+    Compare an interaction with a template interaction or interaction to delete.
+
+    An instance of :class:`Interaction` matches a template instance of the same
+    class or of :class:`DeleteInteraction` if, *a minima*, it involves the same
```

Comment:
*a minima* is confusing, since it's also valid english, and you don't get the italics in source